### ISSUE TITLE
[sort-ngmodule-metadata-arrays] regression creating a false positive on deps

### ISSUE THREAD

**Issue by alfaproject** [angular-eslint/angular-eslint#703]
**Description and reproduction of the issue**

I've just updated to version 12.4.1 from 12.3.1 and `sort-ngmodule-metadata-arrays` started reporting issues on provider deps which as you know can not be sorted safely.

```JSON
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": "error"
  }
}
```

```TS
@NgModule({
  providers: [
    {
      provide: API_LINK,
      useFactory: createHttpLink,
      // eslint-disable-next-line @angular-eslint/sort-ngmodule-metadata-arrays
      deps: [HttpLink, EnvService, Injector, LocaleService],
    },
  ],
})
export class RegressedModule {}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `12.4.1` |
| `@typescript-eslint/parser`        | `4.29.3` |
| `ESLint`                           | `7.32.0` |

```sh
Angular CLI: 12.2.4
Node: 14.17.5
Package Manager: yarn 1.22.5
OS: darwin x64

Angular: 12.2.4
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, material, platform-browser, platform-browser-dynamic
... platform-server, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.4
@angular-devkit/build-angular   12.2.4
@angular-devkit/core            12.2.4
@angular-devkit/schematics      12.2.4
@nguniversal/builders           12.1.0
@nguniversal/express-engine     12.1.0
@schematics/angular             12.2.4
rxjs                            6.6.7
typescript                      4.3.5
webpack                         5.52.1
```

**Comment by CRACKISH** [comment]
+ same error

**Comment by vdumbrav** [comment]
Same error with InjectionToken in Transloco

**Comment by Badisi** [comment]
Also facing the same issue.

@JamesHenry, any chance you could review the [PR](#704) from @rafaelss95 ?
Thanks!
